### ISSUE TITLE
Trim Paths

### ISSUE THREAD

**Issue by Tallowick** [opentoonz/opentoonz#4851]
### Is your feature request related to a problem? Please describe

It would be nice to have the option to anime vector paths by keyframing how much of the line is showing, like the Trim Paths option in After Effects. 

An example where it would be handy is if I want to animate a line being drawn.
I haven't been able to find a way to achieve this effect even indirectly with the current tools and FX that doesn't involve elaborate masks or tweening. 
The vector auto tweening works fine for perfectly straight lines, but for any kind of bezier curve the only viable option is to animate it frame by frame.

### Describe the solution you'd like

An animatable parameter that can be keyeframed as a percentage of the vector line to show would be really convenient (similar to how there is one for the percentage along which a column is along a motion path)
Or alternatively have it listed in the FX lists and when applied to a column the start and end percentages can be adjusted and keyframed. 

For example at 100% the full line is showing
At 50% the line is only drawn up to the middle
And at 0% it is completely at the first node.

A system similar to how there can only be one vector path in a motion path could be used to ensure there aren't multiple paths in such a layer.

**Comment by flurick** [comment]
Many of the vector brushes under the **Generated** section can be tweaked in the palettes **Settings** tab and even keyframe animated in the Palette window.  

It is almost possible already using a very wide Vanishing brush, but a simpler one without the fading part would be great.

**Comment by flurick** [comment]
Yeah, the generated ones are not customizable beyond their settings. 
Good news is that adding an offset slider was easy.

Next challenge is to figure out if it has to be a new brush to not break older projects that maybe where using it.

Other design thingies: 
The other sliders are still relative to the line width, maybe a length % would be good for them also. Or if this is a new brush should all sliders be relative to the lines length? 

[Kooha-2023-04-29-15-39-44.webm](https://user-images.githubusercontent.com/3645797/235306189-b3404644-a419-45a4-bf47-1cd1f8465d98.webm)

https://github.com/opentoonz/opentoonz/commit/0c160d870f98f3101368633ef56b2639efc78433

**Comment by openanim** [comment]
I agree, i strongly support this, to have more parameters for animatable vector styles, a potential that is a bit neglected. In many cases they are useful and quick to set up but there is an obvious lack of parameters available to do exactly what you want. Certainly an approach to applying effects that would be closer to after effects.

I've played with many of them in the past years, and this one also frustrated me quite a bit, for example I couldn't find a viable solution to remove the blur and the stylized text is annoying and imo should be a seperate parameter himself : 

https://user-images.githubusercontent.com/29790512/235352400-ede72d2c-57aa-4391-b279-8e339caf4b67.mp4

https://user-images.githubusercontent.com/29790512/235352707-ea240028-36f3-45c4-9bc9-b88c36df17f8.mp4

Some parameters would help a lot especially for this style:

blur / edge blur : quite frustrating to have the text blurred without being able to remove it
start and end : To choose the direction of the effect (like trimpath)

I had also encountered a problem on the size and length of the strokes to apply the effect: I had to enlarge my text to a certain point so that the effect applies as far as I wanted.
There are also some deformations / sharp shapes due to the style, it would be nice if you could have the control to turn it on and off and have the text or drawing in its original form without it being stylized.

**Comment by flurick** [comment]
I wonder if it was suppose to be rounded since it's real name is dotted-line, hmm
![Screenshot](https://user-images.githubusercontent.com/3645797/235514740-e4b0d7a5-d8c2-4947-8714-7de76b57453b.png)

**Comment by Tallowick** [comment]
>  Good news is that adding an offset slider was easy.
> 

Oh, sweet
How does one go about implementing it? Do you directly edit the source code?

**Comment by RodneyBaker** [comment]
It'd be great to see a follow up on this especially as some of the development was implemented outside of Opentoonz itself.

**Comment by RodneyBaker** [comment]
Tagging with 'roadmap' primarily to suggest additional documentation on working with generated styles would be useful.  Many thanks to @flurick for the exploration done thus far.
